tests/test_utils/__init__.py

```diff
@@ -97,27 +104,44 @@ async def cb(*args, **kwargs):
     return Mock(side_effect=cb)
 
 
-@attr.s
-class FakeResponse:
+# Type ignore: it does not fully implement IResponse, but is good enough for tests
+@zope.interface.implementer(IResponse)
+@attr.s(slots=True, frozen=True, auto_attribs=True)
+class FakeResponse:  # type: ignore[misc]
     """A fake twisted.web.IResponse object
 
     there is a similar class at treq.test.test_response, but it lacks a `phrase`
     attribute, and didn't support deliverBody until recently.
     """
 
-    # HTTP response code
-    code = attr.ib(type=int)
+    verison: Tuple[bytes, int, int] = (b"HTTP", 1, 1)
 
-    # HTTP response phrase (eg b'OK' for a 200)
-    phrase = attr.ib(type=bytes)
```

Comment:
Do we not use this? I'm quite confused by the changes in this file -- they seem unrelated to the rest of the PR.

Reply:
Oh, I see this got moved below.

Still this refactoring seems like it could be split out for a separate review.